service/history/decisionChecker.go

```diff
@@ -191,8 +195,8 @@ func (v *decisionAttrValidator) validateActivityScheduleAttributes(
 		return &workflow.BadRequestError{Message: "ScheduleActivityTaskDecisionAttributes is not set on decision."}
 	}
 
-	if attributes.TaskList == nil || attributes.TaskList.GetName() == "" {
-		return &workflow.BadRequestError{Message: "TaskList is not set on decision."}
+	if err := v.validateTaskListName(attributes.TaskList, ""); err != nil {
```

Comment:
can you set this to be a const `noTasklist` or something?

Reply:
done